### PR TITLE
LOOP-1120 test updates for more complete tests runs

### PR DESCRIPTION
**Pull request by jh-bate**
a lot cleanup for expanded CI tests 

- naming
- greater number of accessibility tests for extra screens that weren't covered
- add test type that the CI build can target 
- fix tests due to UI changes in loop

### CODE REVIEW

File: e2e/functional/functionalTest.spec.js [hunk 1]
```diff
@@ -13,51 +12,19 @@ describe('functional test', () => {
...
-            guardrailsTests.correctionRangeScheduleTests(test);
-        });
-        describe('insulin carb ratio', () => {
-            guardrailsTests.insulinCarbRatioTests(test);
-        });
-        describe('basal rate schedule', () => {
-            guardrailsTests.basalRateScheduleTests(test);
-        });
+    describe('carb entry screen', () => {
+        functionalityTests.carbEntryScreenFunctionalityTests(test);
```
Comment (darinkrauss) [inline e2e/functional/functionalTest.spec.js]: [nit] Redundant "functionality" use here?

File: scripts/test.sh [hunk 2]
```diff
@@ -60,7 +60,12 @@ ln -sf "${BUILD_ROOT}" build
...
   detox test e2e/functional --configuration "${CONFIGURATION}" --loglevel info --record-logs failing --bail --cleanup
-else
+fi
+if [ "${TEST_TYPE}" = "smoke" ]; then
   info "Running smoke tests '${CONFIGURATION}'..."
   detox test e2e/smoke --configuration "${CONFIGURATION}" --loglevel info --record-logs failing --bail --cleanup
 fi
+if [ "${TEST_TYPE}" = "guardrails" ]; then
+  info "Running guardrails tests '${CONFIGURATION}'..."
+  detox test e2e/guardrails --configuration "${CONFIGURATION}" --loglevel info --record-logs failing --bail --cleanup
```
Comment (darinkrauss) [inline scripts/test.sh]: [nit] If test type is going to be a single choice, then consider just:
```
  info "Running ${TEST_TYPE} tests '${CONFIGURATION}'..."
  detox test e2e/${TEST_TYPE} --configuration "${CONFIGURATION}" --loglevel info --record-logs failing --bail --cleanup
```
without the surrounding `if/fi`s.

Comment (darinkrauss) [reply scripts/test.sh]: You could also validate that `TEST_TYPE` is valid (one of `functional`, `smoke`, or `guardrails`) above.

File: tests/accessibility/homeGlucoseScreen.tests.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        await glucoseScreen.Close();
+    });
+
+    it('has a header', async () => {
+        await expect(glucoseScreen.Header()).toExist();
+    });
+    it('has a Carbohydrates Label', async () => {
+        await expect(glucoseScreen.CarbohydratesLabel()).toExist();
+    });
+    it('has a Carbohydrates Label', async () => {
```
Comment (darinkrauss) [inline tests/accessibility/homeGlucoseScreen.tests.js]: [nit] duplicate test?

**Review by darinkrauss — APPROVED** [review]
A couple nits, but LGTM! 🧇

File: e2e/functional/functionalTest.spec.js (hunk 1 above)
Comment (jh-bate) [reply e2e/functional/functionalTest.spec.js]: agreed

File: scripts/test.sh (hunk 2 above)
Comment (jh-bate) [reply scripts/test.sh]: very good point!

**Pull request merged**